src/effects/ChromaKey.cpp

```diff
@@ -67,40 +72,458 @@ void ChromaKey::init_effect_details()
+							pixel[2] *= alphamult;
+							pixel[3] *= alphamult;
+						}
+					}
+				}
+				break;
+
+			case CHROMAKEY_HSV_S:
+			case CHROMAKEY_HSL_S:
+				for (int y = 0; y < height; ++y)
+				{
+					unsigned char *pixel = image->scanLine(y);
+
+					for (int x = 0; x < width; ++x, pixel += 4, pf += 3)
+					{
+						float tmp = fabs(pf[1] - mask.f[1]) * 255;
+
+						if (tmp <= threshold)
+						{
+							pixel[0] = pixel[1] = pixel[2] = pixel[3] = 0;
+						}
+						else if (tmp <= threshold + halothreshold)
+						{
+							float alphamult = (tmp - threshold) / halothreshold;
+
+							pixel[0] *= alphamult;
+							pixel[1] *= alphamult;
+							pixel[2] *= alphamult;
+							pixel[3] *= alphamult;
+						}
+					}
+				}
+				break;
+
+			case CHROMAKEY_HSV_V:
+			case CHROMAKEY_HSL_L:
+				for (int y = 0; y < height; ++y)
+				{
+					unsigned char *pixel = image->scanLine(y);
+
+					for (int x = 0; x < width; ++x, pixel += 4, pf += 3)
+					{
+						float tmp = fabs(pf[2] - mask.f[2]) * 255;
+
+						if (tmp <= threshold)
+						{
+							pixel[0] = pixel[1] = pixel[2] = pixel[3] = 0;
+						}
+						else if (tmp <= threshold + halothreshold)
+						{
+							float alphamult = (tmp - threshold) / halothreshold;
+
+							pixel[0] *= alphamult;
+							pixel[1] *= alphamult;
+							pixel[2] *= alphamult;
+							pixel[3] *= alphamult;
+						}
+					}
+				}
+				break;
+
+			case CHROMAKEY_YCBCR:
+				for (int y = 0; y < height; ++y)
+				{
+					unsigned char *pixel = image->scanLine(y);
+
+					for (int x = 0; x < width; ++x, pixel += 4, pc += 3)
+					{
+						int db = (int) pc[1] - mask.u[1];
+						int dr = (int) pc[2] - mask.u[2];
+						float tmp = sqrt(db * db + dr * dr);
+
+						if (tmp <= threshold)
+						{
+							pixel[0] = pixel[1] = pixel[2] = pixel[3] = 0;
+						}
+						else if (tmp <= threshold + halothreshold)
+						{
+							float alphamult = (tmp - threshold) / halothreshold;
+
+							pixel[0] *= alphamult;
+							pixel[1] *= alphamult;
+							pixel[2] *= alphamult;
+							pixel[3] *= alphamult;
+						}
+					}
+				}
+				break;
+
+			case CHROMAKEY_CIE_LCH_L:
+				for (int y = 0; y < height; ++y)
+				{
+					unsigned char *pixel = image->scanLine(y);
+
+					for (int x = 0; x < width; ++x, pixel += 4, pf += 3)
+					{
+						float tmp = fabs(pf[0] - mask.f[0]);
+
+						if (tmp <= threshold)
+						{
+							pixel[0] = pixel[1] = pixel[2] = pixel[3] = 0;
+						}
+						else if (tmp <= threshold + halothreshold)
+						{
+							float alphamult = (tmp - threshold) / halothreshold;
+
+							pixel[0] *= alphamult;
+							pixel[1] *= alphamult;
+							pixel[2] *= alphamult;
+							pixel[3] *= alphamult;
+						}
+					}
+				}
+				break;
+
+			case CHROMAKEY_CIE_LCH_C:
+				for (int y = 0; y < height; ++y)
+				{
+					unsigned char *pixel = image->scanLine(y);
+
+					for (int x = 0; x < width; ++x, pixel += 4, pf += 3)
+					{
+						float tmp = fabs(pf[1] - mask.f[1]) <= threshold;
```

Comment:
```suggestion
						float tmp = fabs(pf[1] - mask.f[1]);
```

I think I see the problem with LCH Chroma. That comparison shouldn't be there, should it?